tests/python/unittest/test_sparse_operator.py

```diff
@@ -1820,6 +1820,21 @@ def check_scatter_ops(name, shape, lhs_stype, rhs_stype, forward_mxnet_call, for
                           lambda l, r: l + r,
                           rhs_is_scalar=True, verbose=False, density=0.5)
 
+@with_seed()
+def test_mkldnn_sparse():
+    # This test is trying to create a race condition describedd in
```

Comment:
Is this test able to product the race condition reliably or would it be flaky?

Reply:
this test can't produce the race condition. actually, the race condition we think may happen actually can't happen with the current implementation of sparse operators. I may need to wait until the sparse operators are fixed.

I created this PR because I want to address https://github.com/apache/incubator-mxnet/issues/10189. I already have implemented a solution.

Reply:
I'm afraid I don't really understand. Wouldn't this mean that this test is not able to serve its described purpose?

Reply:
You are right. In other words, the race condition actually doesn't exist in the current implementation of MXNet thanks to the bug in sparse operators.

Reply:
I see, thanks for elaborating. What would be the path forward here?

Reply:
I just rename the PR. Given that we can't generate the race condition, I'll just remove any functions that perform in-place data layout conversion, which is the source of the race condition we concern about. In the future (e.g., when the sparse operation bug is fixed), if someone calls data() on an MKLDNN NDArray, the code will print an error message to notify the developer.

Reply:
Sounds great, thanks!